test/macros.jl

```diff
@@ -265,6 +265,40 @@ function macros_test(ModelType::Type{<:JuMP.AbstractModel}, VariableRefType::Typ
         @test con.set == MOI.SecondOrderCone(2)
     end
 
+    @testset "@build_constraint (SOS1)" begin 
+        model = ModelType() 
+        @variable(model, x[1:3]) 
+        con = @build_constraint(x in JuMP.SOS1())
+        con2 = @build_constraint(x in JuMP.SOS1([1,2,3])) 
```

Comment:
So one rule of testing. You want to make it so that the cases are unambiguous. In this case (I know the code is correct) if you passed a vector `[1, 2, 3]` it might still go down the `collect(1:1.0:dim)` path, and your test won't catch it. But if you pass the vector `[4.0, 6.0, 5.0]`, you're testing i) that the numbers don't have to be 1, 2, 3; and ii) that they don't have to be sorted in order.

Reply:
Got it, changing it right now!